### DISCUSSION TITLE
What's the recommended way to inject dependencies like a global singleton services from Angular or Ember?

### DISCUSSION THREAD

**Discussion by niamleeson** [pmndrs/valtio#910] [Q&A]
Hello, thank you for the amazing library. I read through the docs, searched for discussions, and googled, but I haven't found a recommended way to inject dependencies like global singletons like services from Angular or Ember. I saw [valtio-factory](https://github.com/mfellner/valtio-factory/tree/main), but I am not sure if that's ideal. Have you had any discussions around this topic?

**Comment by dai-shi** [comment]
> to inject dependencies like global singletons

Can you explain it with some code snippets?
